sympy/matrices/normalforms.py

```diff
@@ -108,16 +108,14 @@ def clear_row(m):
         if ind:
             m = m.permute_cols([[0, ind[0]]])
 
-    m = clear_column(m)
-    m = clear_row(m)
-    if 1 in m.shape:
-        return [m[0]]
-
-    while not (m[0,1:].is_zero) or not (m[1:,0].is_zero):
+    while any([m[0,i] != 0 for i in range(1,m.cols)]) or any([m[i,0] != 0 for i in range(1,m.rows)]):
```

Comment:
This line is too long but I've been told that line breaks are not encouraged. Not sure how to set it up. Define the lists as variables before the loop and update them in the loop?